### ISSUE TITLE
GHAS/CodeQL reporting missing input sanitization

### ISSUE THREAD

**Issue by automartin5000** [smithy-lang/smithy-typescript#1447]
I posted [this](https://github.com/aws/aws-sdk-js-v3/issues/6623) in the AWS SDK repo, but thought I should repost here since this seems to be the source.

[This code](https://github.com/smithy-lang/smithy-typescript/blob/0d9cee80619a89a6585fa2339c4382084c6036bc/packages/smithy-client/src/quote-header.ts#L6) has started triggering a CodeQL finding for us.

Specifically, it's this line of code:
```typescript
part = `"${part.replace(/"/g, '\\"')}"`;
```

The full security notification is:
> Incomplete string escaping or encoding - This does not escape backslash characters in the input.

**Comment by phuhung273** [comment]
@automartin5000 +1 with you. Currently blocked because of this

**Comment by kuhe** [comment]
responded in https://github.com/aws/aws-sdk-js-v3/issues/6623

**Comment by kuhe** [comment]
See for https://github.com/aws/aws-sdk-js-v3/issues/6623 answer. The CodeQL maintainer recommended dismissing the false positive.
